app/javascript/components/patient/assessment/actions/ExtendedIsolation.js

```diff
@@ -37,130 +37,119 @@ class ExtendedIsolation extends React.Component {
-              <Button
-                variant="secondary btn-square"
-                onClick={() => this.setState({ showExtendIsolationModal: false, extended_isolation: this.props.patient.extended_isolation })}>
-                Cancel
-              </Button>
-              <Button variant="primary btn-square" onClick={this.submit} disabled={this.state.loading}>
-                {this.state.loading && (
-                  <React.Fragment>
-                    <span className="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>&nbsp;
-                  </React.Fragment>
+              </React.Fragment>
+            ) : (
+              <React.Fragment>
+                {this.state.extended_isolation ? (
+                  <Form.Label className="mb-2">
+                    {`Are you sure you want to change the Extend Isolation Date to ${moment(this.state.extended_isolation).format(
+                      'MM/DD/YYYY'
+                    )}? Since this record is on the Closed line list, updating this value will not move this record to another line list. If this individual should be actively monitored, please update the record’s Monitoring Status.`}
+                  </Form.Label>
+                ) : (
+                  <Form.Label className="mb-2">
+                    {`Are you sure you want to clear the "Extend Isolation To" date? Since this record is on the Closed line list, updating this value will not move this record to another line list. If this individual should be actively monitored, please update the record’s Monitoring Status.`}
+                  </Form.Label>
                 )}
-                Submit
-              </Button>
-            </Modal.Footer>
-          </Modal>
-        )}
+              </React.Fragment>
+            )}
+          </Form.Group>
+          <p>Please include any additional details:</p>
+          <Form.Group>
+            <Form.Control
+              as="textarea"
+              rows="2"
+              id="reasoning"
+              onChange={event => this.setState({ reasoning: event.target.value })}
+              aria-label="Additional Details Text Area"
+            />
+          </Form.Group>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            variant="secondary btn-square"
+            onClick={() => this.setState({ showExtendIsolationModal: false, extended_isolation: this.props.patient.extended_isolation })}>
+            Cancel
+          </Button>
+          <Button variant="primary btn-square" onClick={this.submit} disabled={this.state.loading}>
+            {this.state.loading && (
+              <React.Fragment>
+                <span className="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>&nbsp;
+              </React.Fragment>
+            )}
+            Submit
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    );
+  };
+
+  render() {
+    return (
+      <React.Fragment>
```

Comment:
Looks like this modal is broken. It doesn't show up for me and I get this error in the console when I set extended isolation date.
![image](https://user-images.githubusercontent.com/8235974/117191036-1dc07900-adae-11eb-937d-7a3c9713e0bd.png)

Reply:
https://github.com/SaraAlert/SaraAlert/pull/861/commits/fa159f7e664d424ff748a7d0c05555254796382a#diff-6337d7da61c54cfb59fc72e5f27ded4af9bd80a0009c88d7019ddfd16e93d9d3R152

Reply:
Thank you for beefing up the tests too!